web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx

```diff
@@ -49,21 +57,42 @@ export function CreateDatabaseDialog({
   close,
   next,
   dbName,
+  onOverwrite,
+  onTimeout,
 }: CreateDatabaseDialogProps) {
+  /**
+   * Most likely cause of timeout is when we found a matching db_service
+   * but no db_server heartbeats. Most likely cause is because db_service
+   * has been stopped but is not removed from teleport yet (there is some
+   * minutes delay on expiry)
+   */
+  const skipOnTimeout =
+    attempt.status === 'failed' && attempt.statusText === timeoutErrorMsg;
+
   let content: JSX.Element;
-  if (attempt.status === 'failed') {
+  if (attempt.status === 'failed' && !skipOnTimeout) {
```

Comment:
Also, regarding `else if (attempt.status === 'processing')` below: I'm not sure if it matters that much here, but I think it's a good practice to always account for the `''` status as well. In 90% the handling of it is similar to that of `processing`. But the way the conditional in this component is written at the moment means that if `attempt.status` ends up being set from within an effect from `''` to `'processing'`, for a split second we'll end up rendering the "success" branch of the conditional.